src/containers/LessonsContainer/LessonsContainer.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Box, Grid, TextField, Typography } from '@material-ui/core';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import DateFnsUtils from '@date-io/date-fns';
+import { KeyboardDatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { makeStyles } from '@material-ui/core/styles';
+import { fetchGroups } from '../../store/actions/groupsAction';
+import { fetchSubjects } from '../../store/actions/subjectsAction';
+import { getTeachersBySubject } from '../../store/actions/teachersActions';
+import { getWeekdates } from '../../helpers/helpers';
+
+const useStyles = makeStyles(() => ({
+  lessonsData: {
+    margin: 50,
+    paddingLeft: 100,
+    paddingRight: 400,
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  autocomplete: {
+    marginBottom: 25,
+  },
+  dateBox: {
+    textAlign: 'center',
+  },
+  dateText: {
+    marginBottom: 5,
+    textAlign: 'start',
+    marginLeft: 20,
+  },
+  autoComplTeacher: {
+    marginTop: 20,
+  },
+}));
+
+const LessonsContainer = () => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchGroups());
+    dispatch(fetchSubjects());
+  }, [dispatch]);
+
+  const { groups } = useSelector((state) => state.groups);
+  const { subjects } = useSelector((state) => state.subjects);
+  const { teachersBySubject } = useSelector((state) => state.teachers);
+
+  const [group, setGroup] = useState(null);
+  const [subject, setSubject] = useState(null);
+  const [teacher, setTeacher] = useState(null);
+  const [selectedDate, setSelectedDate] = useState(new Date());
```

Comment:
Зачем нам сохранять целые объекты со всеми значениями отдельно по group, subject... нам же нужны только их id, т.е. из всего объекта group только groupId.
Почему нам не сделать один объект и сохранять только те значения, которые нам нужны
```suggestion
const [lesson, setLesson] = useState({
    groupId: '',
    subjectId: '',
    teacherId: '',
  });
```
Также потом через setLesson добавлять нужные переменные в один стейт.

Reply:
Надо еще значение дат добавить

Reply:
исправил